specification/azsadmin/resource-manager/compute/Microsoft.Compute.Admin/preview/2018-07-30-preview/examples/Disks/Get.json

```diff
@@ -0,0 +1,27 @@
+{
+  "parameters": {
+    "subscriptionId": "04666444-56f0-4d4f-afc5-dbd6b134b084",
+    "location": "redmond",
+    "DiskId": "423e4eb4-f791-4d13-ad5b-4d415733b0d6",
+    "api-version": "2018-07-30-preview"
+  },
+    "responses": {
+        "200": {
+            "body": {
+                "id": "/subscriptions/04666444-56f0-4d4f-afc5-dbd6b134b084/providers/Microsoft.Compute.Admin/locations/redmond/disks/423e4eb4-f791-4d13-ad5b-4d415733b0d6",
+                "name": "redmond/423e4eb4-f791-4d13-ad5b-4d415733b0d6",
+                "type": "Microsoft.Compute.Admin/locations/disks",
+                "location": "redmond",
+              "properties": {
+                "diskId": "423e4eb4-f791-4d13-ad5b-4d415733b0d6",
```

Comment:
this value is already in the ID and in name property. Do not repeat it here 3rd time.

Reply:
ID and Name has to be marked as Readonly because of swagger rules. And if it is marked as readonly, it cannot be serialized back to server. So we have to have another property.

Reply:
You should create a new object called DiskMigrationParameters which take all the necessary values you need to do a migration.  Don't include any values you can get doing a disk lookup on the server side.